build.rs

```diff
@@ -42,9 +50,26 @@ fn main() -> io::Result<()> {
 
     // Bland version text
     let mut f = File::create(path).unwrap_or_else(|_| { panic!("{}", path.to_string_lossy().to_string()) });
-    writeln!(f, "{}", strip_codes(&ver))?;
+    writeln!(f, "{}", strip_codes(&ver))
+}
+
+/// Create the perfect hashing for file typing
+fn create_file_typing_hash_file() -> io::Result<()> {
+    let path = &PathBuf::from(env::var("OUT_DIR").unwrap()).join("filetype_maps.rs");
+    let mut file = io::BufWriter::new(File::create(path).unwrap_or_else(|_| { panic!("{}", path.to_string_lossy().to_string()) }));
+    generate_extension_type_map(file.get_mut())?;
+    generate_filename_type_map(file.get_mut())?;
+    file.flush()
+}
 
-    Ok(())
+/// Create the perfect hashing for file icons
```

Comment:
I think you should expand on this, how it's done and the motivation for doing it this way.